docs/docs/getting_started/inputs-and-outputs.mdx

```diff
@@ -8,67 +8,59 @@ The transpiled Cairo contracts have a different ABI than their Solidity counterp
 the Cairo ABI or the Solidity ABI for passing constructor arguments or function parameters. By default, the Solidity ABI
 is used.
 
-Following examples demonstrate how to use both methods
-
 ### Using Solidity ABI
 
-After transpiling the OpenZepplin ERC20 contract, we need to pass two strings to the constructor that
-represent the name and ticker of the token. It can be deployed using Warp as follows:
+The Solidity ABI lets you pass in arguments in a schema similar to what users would use on L1.
+This is supported in Warp's `deploy`, `invoke` and `call` commands.
+Warp takes these inputs and transcodes them into Cairo to forwards them onto the desired network.
 
 ```
-warp deploy warp_output/ERC20__WC__ERC20.cairo --inputs \"NETHERCOIN\",\"NETH\"
+warp invoke <Cario Contract> --inputs \"ExampleString\",\[1,2,3,4\],-1,100.
 ```
 
-For using the Solidity ABI to pass inputs, they must be passed as a comma separated list of numbers, strings and lists.
-For example:
-
-```
-\"NETHERCOIN\",10,\[10,5,4,2\]
-```
+### Using Cairo ABI
 
-The above comma separated list passes three parameters as a string, number and list respectively.
+To use the Cairo ABI to pass inputs, add the `--use_cairo_abi` flag to the `deploy`, `invoke` or `call` command.
 
-### Using Cairo ABI
+Note that there are some nuances that come with using the Cairo ABI:
 
-To use the cairo ABI for passing inputs, the `--use_cairo_abi` flag can be passed to the `deploy`, `invoke` or `call` command.
-Starknet does not support Solidity-style strings. So Warp represents strings as dynamic arrays of bytes. In Cairo ABI, each solidity
-string is represented in two parts, the length of the string, and the value of each byte.
-In the same example of OpenZepplin's ERC20, the two strings can be passed as follows using Cairo ABI:
+- StarkNet does not support Solidity-style strings, so Warp represents strings as dynamic arrays of bytes.
+  In Cairo ABI, each dynamic array is represented in two parts, the length of the string, and the values of the array
+  `<arr_len>,<arr[0],arr[1]...arr[n-1]>`.
+  e.g `ExampleString` -> `13,0x45,0x78,0x61,0x6d,0x70,0x6c,0x65,0x53,0x74,0x72,0x69,0x6e,0x67`
+- Negative numbers will need to be transformed into their twos complement. e.g to pass
+  `-1` into a function that takes `int8`, `-1` needs to be converted to `255` and then passed.
+- Data types that are more than 248 bits wide (`int256`/`uint256`/`bytes32`) will need to be split into their upper and
```

Comment:
To correspond to the given example, can we phrase this instead as "split into their lower and upper 128 bits"?